job.go

```diff
@@ -610,6 +610,16 @@ func WithTags(tags ...string) JobOption {
 	}
 }
 
+// WithIdentifier sets the identifier for the job. The identifier
+// is used to uniquely identify the job and is used for logging
+// and metrics.
+func WithIdentifier(id uuid.UUID) JobOption {
+	return func(j *internalJob) error {
+		j.id = id
```

Comment:
this should check if the id passed in is nil and return an error if it is. also a test would be a grand thing to have. 
- errors go here -> https://github.com/go-co-op/gocron/blob/399ac2807bb8cfb0cade9a73f6a8d48d9ba45f19/errors.go#L41
- error occurred tests here -> https://github.com/go-co-op/gocron/blob/64d6e48dae23488d9a160ae3b1f5070e3c06228a/scheduler_test.go#L502

Reply:
@JohnRoesler ok... however uuid.UUID is a struct and it isn't passed as a pointer/ref, so you can't test it for nil.

Reply:
I meant uuid.Nil, which tests for an uninitialized uuid.

Reply:
@JohnRoesler check now.

Reply:
thank you